### PR TITLE
ParcelableCrdtSingleton

### PR DESCRIPTION
**Pull request by csilvestrini**
Parcelable implementation of CrdtSingleton classes

### CODE REVIEW

File: shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSingleton.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
+package arcs.crdt.parcelables
+
+import android.os.Parcel
+import android.os.Parcelable
+import arcs.common.Referencable
+import arcs.common.ReferenceId
+import arcs.crdt.CrdtSet
+import arcs.crdt.CrdtSingleton
+
+object ParcelableCrdtSingleton {
```
Comment (jasonwyatt) [inline shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSingleton.kt]: KDoc would be good to have here, and on the public classes who are its members.

File: shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSingleton.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+                val values = mutableMapOf<ReferenceId, CrdtSet.DataValue<Referencable>>()
+                val items = parcel.readInt()
+                repeat(items) {
+                    values[requireNotNull(parcel.readString())] =
+                        requireNotNull(
+                            parcel.readTypedObject(ParcelableCrdtSet.DataValue.CREATOR)
+                        ) {
+                            "No DataValue found in parcel when reading ParcelableCrdtSingleton.Data"
+                        }.actual
```
Comment (jasonwyatt) [inline shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSingleton.kt]: You can move the `actual` inside of the requireNotNull:

```kotlin
requireNotNull(parcel.readTypedObject(ParcelableCrdtSet.DataValue.CREATOR)?.actual) {
    "No DataValue found in parcel when reading ParcelableCrdtSingleton.Data"
}
```

File: shells/android/javatests/arcs/crdt/parcelables/ParcelableCrdtSingletonTest.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package arcs.crdt.parcelables
```
Comment (jasonwyatt) [inline shells/android/javatests/arcs/crdt/parcelables/ParcelableCrdtSingletonTest.kt]: Copyright comment?  (noticing they're missing on all of these files, actually..)

File: shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSingleton.kt (hunk 1 above)
Comment (csilvestrini) [reply shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSingleton.kt]: Done

File: shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSingleton.kt (hunk 2 above)
Comment (csilvestrini) [reply shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSingleton.kt]: Done

File: shells/android/javatests/arcs/crdt/parcelables/ParcelableCrdtSingletonTest.kt (hunk 3 above)
Comment (csilvestrini) [reply shells/android/javatests/arcs/crdt/parcelables/ParcelableCrdtSingletonTest.kt]: Done.

I'll have a think about adding a lint check for this...

**Pull request merged**